src/GafferScene/IECoreScenePreview/PrimitiveAlgo.cpp

```diff
@@ -482,83 +482,407 @@ void copyIndices(
 }
 
 
-} // namespace
+struct MergePrimitivesMeshResult
```

Comment:
I like this refactoring. I think it could be even clearer if all the members not needed by `mergePrimitivesInternal()` were private (and `m_` prefixed in the case of member data). Then on first read you'd understand that the public methods (and only the public methods) were the contract.